homeassistant/components/owlet/__init__.py

```diff
@@ -0,0 +1,95 @@
+"""The Owlet Smart Sock integration."""
+from __future__ import annotations
+
+import logging
+
+from pyowletapi.api import OwletAPI
+from pyowletapi.exceptions import (
+    OwletAuthenticationError,
+    OwletConnectionError,
+    OwletEmailError,
+    OwletPasswordError,
+)
+from pyowletapi.sock import Sock
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_TOKEN,
+    CONF_REGION,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_OWLET_EXPIRY, CONF_OWLET_REFRESH, DOMAIN, SUPPORTED_VERSIONS
+from .coordinator import OwletCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Owlet Smart Sock from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    owlet_api = OwletAPI(
+        region=entry.data[CONF_REGION],
+        token=entry.data[CONF_API_TOKEN],
+        expiry=entry.data[CONF_OWLET_EXPIRY],
+        refresh=entry.data[CONF_OWLET_REFRESH],
+        session=async_get_clientsession(hass),
+    )
+
+    try:
+        token = await owlet_api.authenticate()
+
+        if token:
+            hass.config_entries.async_update_entry(entry, data={**entry.data, **token})
```

Comment:
I haven't looked at your underlying api, but why not handle the token inside the api? Cleans up the code here a bit.

Since you are getting a new token everytime you start up, it doesn't really need to be stored in a config entry imo